### DISCUSSION TITLE
[Proto] Migrated from requests-async, missing proto signatures?

### DISCUSSION THREAD

**Discussion by YifeiY** [encode/httpx#1792] [Q&A]
Hi,

I am trying to migrate requests-async code to httpx. requests-async accepts `google.protobuf.pyext._message.ScalarMapContainer` as the type for headers. However httpx doesn't seem to support this. In [httpx._models.py#L850](https://github.com/encode/httpx/blob/master/httpx/_models.py#L850), it expects `k, v` pair which raises `ValueError: too many values to unpack (expected 2)` error.

My questions are,
1. Is it true that httpx dropped the support for proto when migrating from requests_async?
2. Is there a plan to add proto support in a future version?

**Comment by lovelydinosaur** [comment]
We've never intentionally supported those - must have been an accident of the implementation.

You probably want to use `headers=dict(...)` to forcibly coerce it into a proper dict type.

**Reply by YifeiY** [reply]
Ok I see. Thanks

**Reply by YifeiY** [reply]
Is it possible to add proto support in the future?
